lib/ansible/modules/windows/setup.ps1

```diff
@@ -318,11 +318,12 @@ if($gather_subset.Contains('platform')) {
 
     $ansible_facts += @{
         ansible_architecture = $win32_os.OSArchitecture
-        ansible_domain = $ip_props.DomainName
-        ansible_fqdn = ($ip_props.Hostname + "." + $ip_props.DomainName)
-        ansible_hostname = $env:COMPUTERNAME
+        ansible_domain = $win32_cs.Domain.Substring($win32_cs.Workgroup.length)
+        ansible_fqdn = ($win32_cs.DNSHostname + "." + $win32_cs.Domain.Substring($win32_cs.Workgroup.length))
```

Comment:
This still has the same issue as before where a non-domain host would be `HOSTNAME.` (with the ending .). We should conditionally add the domain part with `.` to the FQDN if it's not an empty string. Also the same thing here about the Substring for the domain name.

Reply:
Kept the dot or backward compatibility, so you wish to remove it if not needed?

Reply:
Last commit removes it